src/main/java/com/cloudbees/jenkins/plugins/BitBucketTrigger.java

```diff
@@ -1,114 +1,101 @@
+import com.cloudbees.jenkins.plugins.payload.BitbucketPayload;
+
 import hudson.Extension;
 import hudson.Util;
 import hudson.console.AnnotatedLargeText;
-import hudson.model.Action;
-import hudson.model.CauseAction;
-import hudson.model.Hudson;
-import hudson.model.Item;
-import hudson.model.Job;
+import hudson.model.*;
+import hudson.scm.PollingResult;
 import hudson.triggers.Trigger;
 import hudson.triggers.TriggerDescriptor;
 import hudson.util.SequentialExecutionQueue;
-import hudson.util.StreamTaskListener;
+import jenkins.model.Jenkins;
 import jenkins.model.ParameterizedJobMixIn;
 import jenkins.triggers.SCMTriggerItem;
 import org.apache.commons.jelly.XMLOutput;
 import org.kohsuke.stapler.DataBoundConstructor;
 
 import java.io.File;
 import java.io.IOException;
-import java.io.PrintStream;
 import java.nio.charset.Charset;
-import java.text.DateFormat;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.Date;
-import java.util.logging.Level;
+import java.util.List;
 import java.util.logging.Logger;
 
 
 /**
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 public class BitBucketTrigger extends Trigger<Job<?, ?>> {
+    public List<BitbucketTriggerFilter> triggers;
 
     @DataBoundConstructor
-    public BitBucketTrigger() {
+    public BitBucketTrigger(List<BitbucketTriggerFilter> triggers) {
+        this.triggers = triggers;
     }
 
     /**
      * Called when a POST is made.
      */
-    @Deprecated
-    public void onPost(String triggeredByUser) {
-        onPost(triggeredByUser, "");
-    }
+    public void onPost(final BitbucketEvent bitbucketEvent, final BitbucketPayload bitbucketPayload) {
+        FilterMatcher filterMatcher = new FilterMatcher();
+
+        final List<BitbucketTriggerFilter> matchingFilters = filterMatcher.getMatchingFilters(bitbucketEvent, triggers);
+
+        if(matchingFilters != null) {
+            if(matchingFilters.size() > 0) {
+                BitbucketPollingRunnable bitbucketPollingRunnable = new BitbucketPollingRunnable(job,
+                    getLogFile(),
+                    new BitbucketPollingRunnable.BitbucketPollResultListener() {
+                        @Override
+                        public void onPollSuccess(PollingResult pollingResult) {
+                            for (BitbucketTriggerFilter filter : matchingFilters) {
+                                BitbucketTriggerCause cause;
+                                try {
+                                    cause = filter.getCause(getLogFile(), bitbucketPayload);
+
+                                    if (filter.shouldScheduleJob(bitbucketPayload)) {
+                                        scheduleJob(cause, bitbucketPayload);
+                                        return;
+                                    }
+
+                                } catch (IOException e) {
+                                    e.printStackTrace();
```

Comment:
@StFS I think you are right, I will change it and add it soon to the PR. Any other thing you think should be changed please let me know. 
Thanks!